### PR TITLE
[controller] backupVersionRetentionMs Validation

### PR DESCRIPTION
**Pull request by KaiSernLim**
## Problem Statement

When `backupVersionRetentionMs` is set to a dangerously low value (whether through bugs, tool resets, or human error), backup versions can be deleted before an incident is detected — eliminating the ability to roll back.

## Solution

Add input validation for `backupVersionRetentionMs`, rejecting values that are neither `-1` (cluster default) nor at least 1 day.

Validation is added in two places for defense in depth:
1. **`VeniceParentHelixAdmin.updateStore()`** — catches bad values before propagating to child controllers
2. **`VeniceHelixAdmin.setBackupVersionRetentionMs()`** — catches bad values on direct child controller updates

### Config validation rule

| Config | Rule |
|---|---|
| `backupVersionRetentionMs` | Must be `-1` (cluster default) or `≥ 1 day` |

A new constant `Store.MIN_BACKUP_VERSION_RETENTION_MS = TimeUnit.DAYS.toMillis(1)` is added to the `Store` interface.

### Code changes
- [ ] Added new code behind **a config**. If so list the config names and their default values in the PR description.
- [ ] Introduced new **log lines**.
  - [ ] Confirmed if logs need to be **rate limited** to avoid excessive logging.

### **Concurrency-Specific Checks**
Both reviewer and PR author to verify
- [ ] Code has **no race conditions** or **thread safety issues**.
- [ ] Proper **synchronization mechanisms** (e.g., \`synchronized\`, \`RWLock\`) are used where needed.
- [ ] No **blocking calls** inside critical sections that could lead to deadlocks or performance degradation.
- [ ] Verified **thread-safe collections** are used (e.g., \`ConcurrentHashMap\`, \`CopyOnWriteArrayList\`)
- [ ] Validated proper exception handling in multi-threaded code to avoid silent thread termination.

## How was this PR tested?

- Added \`TestVeniceParentHelixAdmin.testUpdateStoreConfigValidation\` — covers valid boundary values (\`-1\`, exactly 1 day) and invalid values below the minimum, verifies HTTP 400 + \`INVALID_CONFIG\` error type
- Updated \`VeniceParentHelixAdminTest\` integration test to use 1-day retention (meets the new minimum)
- Full \`TestVeniceParentHelixAdmin\` suite passes (0 failures)
- SpotBugs clean, diff coverage above threshold

### CODE REVIEW

File: services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceParentHelixAdmin.java [hunk 1]
```diff
@@ -3067,9 +3090,25 @@ public void updateStore(String clusterName, String storeName, UpdateStoreQueryPa
           largestUsedRTVersionNumber.map(addToUpdatedConfigList(updatedConfigsList, LARGEST_USED_RT_VERSION_NUMBER))
               .orElse(null);
 
+      if (backupVersionRetentionMs.isPresent()) {
+        long retentionMs = backupVersionRetentionMs.get();
+        if (retentionMs >= 0 && retentionMs < Store.MIN_BACKUP_VERSION_RETENTION_MS) {
+          String errorMessage = errorMessagePrefix + "Backup version retention time: " + retentionMs
+              + "ms is below the minimum allowed value of " + Store.MIN_BACKUP_VERSION_RETENTION_MS
+              + "ms (1 day). Set to -1 to use the cluster default retention.";
```
[the remark this answers is not in the document]
Comment (KaiSernLim) [reply services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceParentHelixAdmin.java]: Fixed. Changed `retentionMs >= 0 &&` to `retentionMs != -1 &&` so values like `-2` are now rejected. Only `-1` (cluster default) or `>= MIN_BACKUP_VERSION_RETENTION_MS` are accepted.

File: services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java [hunk 2]
```diff
@@ -5682,6 +5707,13 @@ void setHybridStoreDiskQuotaEnabled(String clusterName, String storeName, boolea
   }
 
   private void setBackupVersionRetentionMs(String clusterName, String storeName, long backupVersionRetentionMs) {
+    if (backupVersionRetentionMs >= 0 && backupVersionRetentionMs < Store.MIN_BACKUP_VERSION_RETENTION_MS) {
+      throw new VeniceHttpException(
+          HttpStatus.SC_BAD_REQUEST,
+          "Backup version retention time: " + backupVersionRetentionMs + "ms is below the minimum allowed value of "
+              + Store.MIN_BACKUP_VERSION_RETENTION_MS + "ms (1 day). Set to -1 to use the cluster default retention.",
+          ErrorType.INVALID_CONFIG);
+    }
```
[the remark this answers is not in the document]
Comment (KaiSernLim) [reply services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java:5690]: Fixed. Same guard change applied: `backupVersionRetentionMs >= 0 &&` → `backupVersionRetentionMs != -1 &&`.

File: services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java [hunk 3]
```diff
@@ -5561,6 +5561,12 @@ void setSeparateRealTimeTopicEnabled(String clusterName, String storeName, boole
...
 
   private void setReplicationFactor(String clusterName, String storeName, int replicaFactor) {
+    if (replicaFactor < 1) {
+      throw new VeniceHttpException(
+          HttpStatus.SC_BAD_REQUEST,
+          "replicationFactor: " + replicaFactor + " must be at least 1.",
+          ErrorType.INVALID_CONFIG);
+    }
     storeMetadataUpdate(clusterName, storeName, (store, resources) -> {
       store.setReplicationFactor(replicaFactor);
```
[the remark this answers is not in the document]
Comment (KaiSernLim) [reply services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java:5566]: Fixed. Added `amplificationFactor < 1` validation in `VeniceHelixAdmin.updateStore()` before the partitioner config merge block, consistent with the parent-side guard.

File: services/venice-controller/src/test/java/com/linkedin/venice/controller/TestVeniceParentHelixAdmin.java [hunk 4]
```diff
@@ -3542,6 +3542,100 @@ private Store setupForStoreViewConfigUpdateTest(String storeName) {
...
+            .updateStore(clusterName, storeName, new UpdateStoreQueryParams().setBackupVersionRetentionMs(0L)));
+
+    // backupVersionRetentionMs: -1 is valid (cluster default)
+    parentAdmin.updateStore(clusterName, storeName, new UpdateStoreQueryParams().setBackupVersionRetentionMs(-1L));
+
+    // backupVersionRetentionMs: exactly 1 day is valid
+    parentAdmin.updateStore(
+        clusterName,
+        storeName,
+        new UpdateStoreQueryParams().setBackupVersionRetentionMs(Store.MIN_BACKUP_VERSION_RETENTION_MS));
```
[the remark this answers is not in the document]
Comment (KaiSernLim) [reply services/venice-controller/src/test/java/com/linkedin/venice/controller/TestVeniceParentHelixAdmin.java]: Fixed. Added a test case for `backupVersionRetentionMs = -2` expecting HTTP 400 + `INVALID_CONFIG`.

File: services/venice-controller/src/test/java/com/linkedin/venice/controller/TestVeniceHelixAdminWithoutCluster.java [hunk 5]
```diff
@@ -313,4 +319,95 @@ public void testSourceRegionSelectionForTargetedRegionPush() {
...
+      Assert.assertTrue(e.getCause() instanceof VeniceHttpException);
+    }
+
+    // -1 (cluster default) should NOT throw (no call to storeMetadataUpdate expected since admin is a mock)
+    // The validation passes; storeMetadataUpdate is mocked to do nothing
+    setBackupRetention.invoke(admin, "cluster", "store", -1L);
+
+    // Exactly 1 day should NOT throw
+    setBackupRetention.invoke(admin, "cluster", "store", Store.MIN_BACKUP_VERSION_RETENTION_MS);
+
```
[the remark this answers is not in the document]
Comment (KaiSernLim) [reply services/venice-controller/src/test/java/com/linkedin/venice/controller/TestVeniceHelixAdminWithoutCluster.java]: Fixed. Added a `-2` case in the reflection-based setter test, verifying HTTP 400 + `INVALID_CONFIG`.

File: services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java [hunk 6]
```diff
@@ -5683,6 +5683,11 @@ void setHybridStoreDiskQuotaEnabled(String clusterName, String storeName, boolea
   }
 
   private void setBackupVersionRetentionMs(String clusterName, String storeName, long backupVersionRetentionMs) {
+    if (backupVersionRetentionMs != -1 && backupVersionRetentionMs < Store.MIN_BACKUP_VERSION_RETENTION_MS) {
+      throw new VeniceException(
+          "Backup version retention time: " + backupVersionRetentionMs + "ms is below the minimum allowed value of "
+              + Store.MIN_BACKUP_VERSION_RETENTION_MS + "ms (1 day). Set to -1 to use the cluster default retention.");
```
[the remark this answers is not in the document]
Comment (KaiSernLim) [reply services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java:5689]: yeah but no
